src/filter.c

```diff
@@ -128,50 +139,70 @@ static bool tile_check_read(char* r1_header, char* r1_seq, char* r1_strand, char
         comp = tiles_to_remove[i];
     }
     
-    free(_fq_header);
+    free(_read_id);
     return ret_val;
 }
 
 
-bool (*check_func)(char*, char*, char*, char*, char*, char*, char*, char*) = std_check_read;
+static bool id_check_read(FastqReadPair read_pair) {
+    char* _read_id = malloc(sizeof (char) * (strlen(read_pair.r1.header) + 1));
+    strcpy(_read_id, read_pair.r1.header);
+    
+    char* coord_information = strtok(_read_id, " ");
+    
+    int i = 0;
+    char* comp = reads_to_remove[i];
+    bool ret_val = true;
+    while(comp != NULL) {
+        if (strcmp(comp, coord_information) == 0) {
+            ret_val = false;
+        }
+        i++;
+        comp = reads_to_remove[i];
+    }
```

Comment:
I guess you'll test on a big file how fast this implementation is going to be.
but it might be worth looking into already implemented hashtables. 
I found one called [uthash](http://troydhanson.github.io/uthash/) which seems relatively straightforward to use.
It supports key as [string](http://troydhanson.github.io/uthash/userguide.html#_string_keys)